python/ros_interface.py

```diff
@@ -0,0 +1,107 @@
+        self.gps_headingDegrees = 0
+        self.gps_speedKmph = 0
+
+        self.past_wind_speedKmph = 0
+        self.past_wind_direction = 0
+        self.past_gps_lat = 0
+        self.past_gps_lon = 0
+        self.past_gps_headingDegrees = 0
+        self.past_gps_speedKmph = 0
+
+    def sensorsCallback(self, data):
+        self.data = data
+
+    def pub(self):
+        self.translate()
+        self.pubGPS.publish(self.gps_lat, self.gps_lon, self.gps_headingDegrees, self.gps_speedKmph)
+        self.pubWind.publish(self.wind_direction, self.wind_speedKmph)
+
+    # Translate from many sensor outputs to one signal, checking error margin, averaging, and converting units
+    def translate(self):
+        # Past values
+        self.past_gps_lat = self.gps_lat
+        self.past_gps_lon = self.gps_lon
+        self.past_gps_headingDegrees = self.gps_headingDegrees
+        self.past_gps_speedKmph = self.gps_speedKmph
+        self.past_wind_speedKmph = self.wind_speedKmph
+        self.past_wind_direction = self.wind_direction
+
+        # GPS sensors
+        self.gps_lat = self.checkErrAndAvg([self.past_gps_lat,
+                                            self.data.gps_0_latitude, self.data.gps_1_latitude])
+        self.gps_lon = self.checkErrAndAvg([self.past_gps_lon,
+                                            self.data.gps_0_longitude, self.data.gps_1_longitude])
+        self.gps_headingDegrees = self.checkErrAndAvg([self.past_gps_headingDegrees,
+                                                       self.convertDegrees(self.data.gps_0_true_heading),
+                                                       self.convertDegrees(self.data.gps_1_true_heading)])
+        self.gps_speedKmph = self.checkErrAndAvg([self.past_gps_speedKmph,
+                                                  self.data.gps_0_groundspeed * KNOTS_TO_KMPH,
+                                                  self.data.gps_1_groundspeed * KNOTS_TO_KMPH])
+        # Wind sensors
+        self.wind_speedKmph = self.checkErrAndAvg([self.past_wind_speedKmph,
+                                                   self.data.wind_sensor_0_speed * KNOTS_TO_KMPH,
+                                                   self.data.wind_sensor_1_speed * KNOTS_TO_KMPH,
+                                                   self.data.wind_sensor_2_speed * KNOTS_TO_KMPH])
+        self.wind_direction = self.checkErrAndAvg([self.past_wind_direction,
+                                                   self.data.wind_sensor_0_direction,
+                                                   self.data.wind_sensor_1_direction,
+                                                   self.data.wind_sensor_2_direction])
+
+    # (0 is north, 90 is east, etc - cw - degrees) -> (0 is east, 90 is north, etc - ccw - degrees)
+    def convertDegrees(self, degree):
+        converted = -1 * degree + 90
+        return converted if converted >= 0 else converted + 360
+
+    def checkErrAndAvg(self, arr):
```

Comment:
Avoid the division by 0. For now, can do a check:

if len == 0:
WARN: Non matching values for GPS, defaulting to previous (or GPS 0)
Then return the first one or something.